R/load.r

```diff
@@ -148,12 +151,20 @@ load_all <- function(path = ".", reset = TRUE, recompile = FALSE,
     unload_dll(package)
   }
 
-  if (recompile) {
-    pkgbuild::clean_dll(path)
+  # Compile dll if requested
+  if (!missing(recompile) && missing(compile)) {
+    compile <- if (isTRUE(recompile)) TRUE else NA
   }
 
-  # Compile dll if it exists
-  pkgbuild::compile_dll(path, quiet = quiet)
+  if (isTRUE(compile)) {
+    pkgbuild::compile_dll(path, force = TRUE, quiet = quiet)
+  } else if (identical(compile, NA)) {
+    pkgbuild::compile_dll(path, quiet = quiet)
+  } else if (isFALSE(compile)) {
```

Comment:
`isFALSE()` is only available in R > 3.5, so we would need an implementation of it in utils or a backport.

Reply:
I used `identical()` which should be fine for this usage.